### PR TITLE
start ReadText deprecation process

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Fixes
* part of #15475

## Approach
In the commit description

## How Has This Been Tested?

[tts.webm](https://github.com/ankidroid/Anki-Android/assets/69634269/01767f92-00ea-4aa4-bd28-9c6a3d64cb24)

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/res/values/10-preferences.xml [hunk 1]
```diff
@@ -283,6 +283,8 @@ this formatter is used if the bind only applies to both the question and the ans
 
     <!-- Advanced -->
     <string name="pref_cat_experimental" maxLength="41" tools:ignore="UnusedResources">Experimental</string>
+    <string name="readtext_deprecation_warn">AnkiDroid has introduced a better TTS mechanism which is compatible with other Anki clients and includes more voices and even language speed!\n\nPlease migrate as soon as possible, as this setting will be removed in the future</string>
+    <string name="readtext_reviewer_warn">Please migrate to the new TTS system</string>
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/main/res/values/10-preferences.xml:287]:
```diff
- migrate
+ upgrade
```

File: AnkiDroid/src/main/java/com/ichi2/anki/ReadText.kt [hunk 2]
```diff
@@ -288,9 +289,10 @@ object ReadText {
                 Timber.w("TTS not successfully initialized")
             }
         }
-        // Show toast that it's getting initialized, as it can take a while before the sound plays the first time
-        if (ankiActivityContext != null) {
-            ankiActivityContext.showSnackbar(R.string.initializing_tts)
+        ankiActivityContext?.showSnackbar(R.string.readtext_reviewer_warn, LENGTH_INDEFINITE) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/ReadText.kt:292]: a comment would be useful

File: AnkiDroid/src/main/res/values/10-preferences.xml (hunk 1 above)
Comment (david-allison) [inline AnkiDroid/src/main/res/values/10-preferences.xml:287]:
```suggestion
    <string name="readtext_reviewer_warn">Please upgrade to the new text to speech format</string>
```

**Review by david-allison — APPROVED** [review]
All good, thanks!

Slight note about `TTS` being abbreviated. 

I do feel that in the reviewer we'll want to spell it out, as a user might have forgotten (and it means translators will not struggle to translate it)

File: AnkiDroid/src/main/res/values/10-preferences.xml [hunk 3]
```diff
@@ -283,6 +283,8 @@ this formatter is used if the bind only applies to both the question and the ans
 
     <!-- Advanced -->
     <string name="pref_cat_experimental" maxLength="41" tools:ignore="UnusedResources">Experimental</string>
+    <string name="readtext_deprecation_warn">AnkiDroid has introduced a better TTS mechanism which is compatible with other Anki clients and includes more voices and even language speed!\n\nPlease migrate as soon as possible, as this setting will be removed in the future</string>
```
Comment (lukstbit) [inline AnkiDroid/src/main/res/values/10-preferences.xml:286]:
```suggestion
    <string name="readtext_deprecation_warn">AnkiDroid has introduced a better TTS mechanism which is compatible with other Anki clients and includes more voices and improvements to language playback!\n\nPlease migrate as soon as possible, as this setting will be removed in the near future</string>
```

**Review by lukstbit — APPROVED** [review]

**Pull request merged**